ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala

```diff
@@ -325,15 +325,15 @@ class ContractsService(
               resolveTemplateId(lc)(jwt, ledgerId)(templateId).map(_.toOption.flatten)
             )
             res <- OptionT(unsafeRunAsync {
-              import doobie.implicits._, cats.syntax.apply._
               timed(
                 metrics.daml.HttpJsonApi.Db.fetchByIdFetch,
-                fetch.fetchAndPersist(jwt, ledgerId, parties, List(resolved)),
-              ) *>
-                timed(
-                  metrics.daml.HttpJsonApi.Db.fetchByIdQuery,
-                  ContractDao.fetchById(parties, resolved, contractId),
-                )
+                fetch.fetchAndPersistBracket(jwt, ledgerId, parties, List(resolved)) { _ =>
+                  timed(
```

Comment:
Not sure about nested `timed` here, how does it work? is it fine?

Reply:
The trouble is that what we are wanting to measure here is the difference between time fetching for the query, and time actually doing the query (i.e. the fetchById call), and conflating the two would be _really bad_ for the accuracy of our metrics. The best solution here might be to make `fetchAndPersistBracket` a little more complicated, possibly by having it take a polymorphic function `ConnectionIO ~> ConnectionIO` as an argument (it seems worse to have it talk about metrics itself).